From 20a81c2d3f0b1b7c456941e063f6b7940164f65d Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Sat, 15 Mar 2014 07:43:05 +0100
Subject: [PATCH 38/43] MINOR: protect ebimtree/ebistree against multiple
 inclusions

These two files were missing the usual #ifndef around the code.
(cherry picked from commit f2873b23705ff08ca3ab4f1a6b27050e572e1d9e)
---
 ebtree/ebimtree.h | 5 +++++
 ebtree/ebistree.h | 4 ++++
 2 files changed, 9 insertions(+)

diff --git a/ebtree/ebimtree.h b/ebtree/ebimtree.h
index 1fe9fa0..99e7503 100644
--- a/ebtree/ebimtree.h
+++ b/ebtree/ebimtree.h
@@ -18,6 +18,9 @@
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
  */
 
+#ifndef _EBIMTREE_H
+#define _EBIMTREE_H
+
 #include <string.h>
 #include "ebtree.h"
 #include "ebpttree.h"
@@ -317,3 +320,5 @@ __ebim_insert(struct eb_root *root, struct ebpt_node *new, unsigned int len)
 	root->b[side] = eb_dotag(&new->node.branches, EB_NODE);
 	return new;
 }
+
+#endif /* _EBIMTREE_H */
diff --git a/ebtree/ebistree.h b/ebtree/ebistree.h
index a1fbb76..e1bcbc7 100644
--- a/ebtree/ebistree.h
+++ b/ebtree/ebistree.h
@@ -20,6 +20,9 @@
 
 /* These functions and macros rely on Multi-Byte nodes */
 
+#ifndef _EBISTREE_H
+#define _EBISTREE_H
+
 #include <string.h>
 #include "ebtree.h"
 #include "ebpttree.h"
@@ -323,3 +326,4 @@ __ebis_insert(struct eb_root *root, struct ebpt_node *new)
 	return new;
 }
 
+#endif /* _EBISTREE_H */
-- 
1.8.3.2

